arches/app/models/migrations/9944_update_refresh_relationship_function.py

```diff
@@ -0,0 +1,150 @@
+                END,
+                '',
+                relationship->>'ontologyProperty',
+                resource_id,
+                (relationship->>'resourceId')::uuid,
+                relationship->>'inverseOntologyProperty',
+                tile_id,
+                nodeid,
+                graphid_from,
+                graphid_to,
+                now(),
+                now()
+            FROM relationships;
+
+            RETURN true;
+        END;
+        $$ language plpgsql;
+    """,
+
+    revert_refresh_tile_resource_relationship_function = """
+        create or replace function __arches_refresh_tile_resource_relationships(
+            tile_id uuid
+        ) returns boolean as $$
+        declare
+            resource_id uuid;
+        begin
+            select resourceinstanceid into resource_id from tiles where tileid = tile_id;
+
+            delete from resource_x_resource where tileid = tile_id;
+
+            with relationships as (
+                select n.nodeid,
+                    jsonb_array_elements(t.tiledata->n.nodeid::text) as relationship
+                from tiles t
+                    left join nodes n on t.nodegroupid = n.nodegroupid
+                where n.datatype in ('resource-instance-list', 'resource-instance')
+                    and t.tileid = tile_id
+                    and t.tiledata->>n.nodeid::text is not null
+            )
+            insert into resource_x_resource (
+                resourcexid,
+                notes,
+                relationshiptype,
+                resourceinstanceidfrom,
+                resourceinstanceidto,
+                inverserelationshiptype,
+                tileid,
+                nodeid,
+                created,
+                modified
+            ) select
+                CASE relationship->>'resourceXresourceId'
+                    WHEN '' THEN uuid_generate_v4()
+                    ELSE (relationship->>'resourceXresourceId')::uuid
+                END,
+                '',
+                relationship->>'ontologyProperty',
+                resource_id,
+                (relationship->>'resourceId')::uuid,
+                relationship->>'inverseOntologyProperty',
+                tile_id,
+                nodeid,
+                now(),
+                now()
+            from relationships;
+
+            return true;
+        end;
+        $$ language plpgsql;
+    """
+
+    update_previous_relationship = """
+        UPDATE resource_x_resource x
+        SET resourceinstancefrom_graphid = r.graphid
+        FROM resource_instances r
+        WHERE r.resourceinstanceid = x.resourceinstanceidfrom
+        AND x.resourceinstancefrom_graphid is null;
+
+        UPDATE resource_x_resource x
+        SET resourceinstanceto_graphid = r.graphid
+        FROM resource_instances r
+        WHERE r.resourceinstanceid = x.resourceinstanceidto
+        AND x.resourceinstanceto_graphid is null;
+    """
+
+    revert_previous_relationship = """
```

Comment:
Because this is an empty string instead of SQL, it causes an error when a user tries to reverse the migration. To do nothing, on reversal, it could be something like `SELECT 1=1`.